content/specs/auth.md

```diff
@@ -0,0 +1,401 @@
+- `grant_types` (array of strings, required): `authorization_code` must always be included. `refresh_token` is optional, but must be included if the client will make token refresh requests.
+- `scope` (string, sub-strings space-separated, required): all scope values which *might* be requested by this client are declared here. The `atproto` scope is required, so must be included here. See “Scopes” section.
+- `response_types` (array of strings, required): `code` must be included. `implicit` is not allowed.
+- `redirect_uris` (array of strings, required): at least one redirect URI is required. See Authorization Request Fields section for rules about redirect URIs, which also apply here.
+- `token_endpoint_auth_method` (string, optional): confidential clients must set this to  `private_key_jwt`.
+- `token_endpoint_auth_signing_alg` (string, optional): `none` is never allowed here. The current recommended and most-supported algorithm is `ES256`, but this may evolve over time. Authorization Servers will compare this against their supported algorithms.
+- `dpop_bound_access_tokens` (boolean, required): DPoP is mandatory for all clients, so this must be present and `true`
+- `jwks` (object with array of JWKs, optional): confidential clients must supply at least one public key in JWK format for use with JWT client authentication. Either this field or the `jwks_uri` field must be provided for confidential clients, but not both.
+- `jwks_uri` (string, optional): URL pointing to a JWKS JSON object. See `jwks` above for details.
+
+These fields are optional but recommended:
+
+- `client_name` (string, optional): human-readable name of the client
+- `client_uri` (string, optional): not to be confused with `client_id`, this is a homepage URL for the client. If provided, the `client_uri` must have the same hostname as `client_id`.
+- `logo_uri` (string, optional): URL to client logo. HTTPS is required over HTTP.
+- `tos_uri` (string, optional): URL to human-readable terms of service (ToS) for the client. HTTPS is required over HTTP.
+- `policy_uri` (string, optional): URL to human-readable privacy policy for the client. HTTPS is required over HTTP.
+
+See "OAuth Security Considerations" below for when `client_name`, `client_uri`, and `logo_uri` will or will not be displayed to end users.
+
+Additional optional client metadata fields are enumerated with [IANA](https://www.iana.org/assignments/oauth-parameters/oauth-parameters.xhtml#client-metadata). Note that these are shared with the "Dynamic Client Registration" standard, which is not used directly by the atproto OAuth profile.
+
+#### Localhost Client Development
+
+When working on client software, it may be difficult for developers to publish in-progress client metadata at a public URL so that authorization servers can access it. This may even be true for development environments using a containerized Authorization Server and local DNS, because of SSRF protections against local IP ranges.
```

Comment:
Not only SSRF protection but also: a server cannot contact another computer's localhost!

```suggestion
When working on client software, it may be difficult for developers to publish in-progress client metadata at a public URL so that authorization servers can access it. This may even be true for development environments using a containerized Authorization Server and local DNS, because there is no way an Authorization Server can interact with a client's loopback interface.
```

Reply:
The specific situation this is describing is where a developer is running the entire atproto stack locally, including the Authorization Server, in docker containers. instead of developing against the live network. in that situation the AS could make localhost connections.

Reply:
There is also the scenario of a client dev working against a prod deployment, which is likely the most common case, don't you think ?

Reply:
updated text